packages/shared/src/hooks/auth/useSignBack.ts

```diff
@@ -0,0 +1,44 @@
+import { useMemo } from 'react';
+import usePersistentContext from '../usePersistentContext';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+import { SocialProvider } from '../../components/auth/common';
+import { LoggedUser } from '../../lib/user';
+
+export const SIGNIN_METHOD_KEY = 'signin_method';
+
+export const signBackKey = generateQueryKey(
+  RequestKey.Auth,
+  null,
+  SIGNIN_METHOD_KEY,
+).join(':');
```

Comment:
Why do we use the join here? 👀

Reply:
It's a react query key, used for something else, I would suggest either making a new function or finding another solution.
Try to keep things seperate

Reply:
This is also a key to be used for storage. What it accepts is a string and to keep the keys uniform, reused the function.

Reply:
Maybe we can duplicate the function and name it explicitly and do the join in the function?
Will be more uniform moving forward for storage keys.

WDYT?

Reply:
Good point! Let's do that.